src/libshadertrap/include/libshadertrap/gl_functions.h

```diff
@@ -0,0 +1,391 @@
+// Copyright 2021 The ShaderTrap Project Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Automatically-generated file - DO NOT EDIT
+
+#ifndef LIBSHADERTRAP_GL_FUNCTIONS_H
+#define LIBSHADERTRAP_GL_FUNCTIONS_H
+
+#include <GLES3/gl32.h>
+
+#include <functional>
+
+namespace shadertrap {
+
+struct GlFunctions {
+  // clang-format off
+  std::function<void(GLuint, GLuint)> glActiveShaderProgram_;
```

Comment:
You're right, but the reason I did this was because I need to give them a different name from their standard name, otherwise when we try to populate the fields in the presence of GLAD things don't work because GLAD has #defined each GL function name to be something different.

Happy to use a different convention.